x/pairing/keeper/unresponsive_provider.go

```diff
@@ -242,76 +166,36 @@ func (k Keeper) getCurrentProviderStakeStorageList(ctx sdk.Context) []epochstora
 	return stakeStorageList
 }
 
-// Function that returns a list of pointers to ProviderPaymentStorage objects by iterating on all the staked providers
-func (k Keeper) getProviderPaymentStorageFromStakeStorageList(ctx sdk.Context, epoch uint64, stakeStorageList []epochstoragetypes.StakeStorage) ([]*types.ProviderPaymentStorage, error) {
-	var providerPaymentStorageList []*types.ProviderPaymentStorage
-
-	// Go over the provider list
-	for _, stakeStorage := range stakeStorageList {
-		for _, stakeEntry := range stakeStorage.GetStakeEntries() {
-			sdkStakeEntryProviderAddress, err := sdk.AccAddressFromBech32(stakeEntry.Address)
-			if err != nil {
-				// if bad data was given, we cant parse it so we ignore it and continue this protects from spamming wrong information.
-				utils.LavaFormatError("unable to sdk.AccAddressFromBech32(provider)", err, &map[string]string{"provider_address": stakeEntry.Address})
-				continue
-			}
-
-			// get providerPaymentStorageKey from the stake entry details
-			providerPaymentStorageKey := k.GetProviderPaymentStorageKey(ctx, stakeEntry.GetChain(), epoch, sdkStakeEntryProviderAddress)
-
-			// get providerPaymentStorage with providerPaymentStorageKey
-			providerPaymentStorage, found := k.GetProviderPaymentStorage(ctx, providerPaymentStorageKey)
-			if !found {
-				continue
-			}
+// Function that punishes providers. Current punishment is unstake
+func (k Keeper) punishUnresponsiveProvider(ctx sdk.Context, providerPaymentStorageKeyList []string) error {
+	// extract from a providerPaymentStorageKey the provider address and the chain ID
+	providerAddress := k.getProviderAddressFromProviderPaymentStorageKey(providerPaymentStorageKeyList[0])
```

Comment:
what if the list is empty?

Reply:
modified the check there. should be ok now